runtime/expr/function/ztests/regexp-replace.yaml

```diff
@@ -0,0 +1,13 @@
+zed: "yield regexp_replace(in, re, new)"
+
+input: |
+  {in:"-ab-axxb-",re:"a(x*)b",new:"T"}
+  {in:"-ab-axxb-",re:"a(x*)b",new:"$1"}
+  {in:"Foo bar",re:"Foo",new:"foo"}
+  {in: "seafood fool",re:"foo(.?",new:"food"}
```

Comment:
We should test null and non-string values for in, re, and new.

Reply:
I don't think this handles either in the way you would expect. Will fix.